src/Widgets/ZoomSliderAssembly.vala

```diff
@@ -0,0 +1,150 @@
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: David Hewitt <davidmhewitt@gmail.com>
+*/
+
+public class ZoomSliderAssembly : Gtk.Grid {
+    private Gtk.Scale slider;
+    private Gtk.EventBox zoom_out_box;
+    private Gtk.EventBox zoom_in_box;
+    private Gtk.EventBox active_zoom_widget;
+    private uint zoom_click_id = 0;
+    private uint zoom_initial_wait_id = 0;
+    private Gtk.Settings settings;
+
+    public signal void value_changed ();
+
+    public string tooltip {
+        set {
+            slider.tooltip_text = value;
+        }
+    }
+
+    public double zoom_value {
+        get {
+            return slider.get_value ();
+        }
+        set {
+            slider.set_value (value);
+        }
+    }
+
+    construct {
+        orientation = Gtk.Orientation.HORIZONTAL;
+        margin_top = 5;
+        margin_bottom = 5;
+    }
+
+    public ZoomSliderAssembly (double min, double max, double step, double initial_val) {
+        settings = get_settings ();
+
+        var zoom_out = new Gtk.Image.from_icon_name (Resources.ICON_ZOOM_OUT, Gtk.IconSize.MENU);
+        zoom_out_box = new Gtk.EventBox ();
+        zoom_out_box.above_child = true;
+        zoom_out_box.visible_window = false;
+        zoom_out_box.add (zoom_out);
+        zoom_out_box.button_press_event.connect (on_zoom_button_pressed);
+        zoom_out_box.button_release_event.connect (on_zoom_button_released);
+        zoom_out_box.leave_notify_event.connect (() => {
+            clear_zoom_timeouts ();
+
+            return false;
+        });
+
+        add (zoom_out_box);
+
+        slider = new Gtk.Scale (Gtk.Orientation.HORIZONTAL, new Gtk.Adjustment (initial_val, min, max, step, step, 0));
+        slider.value_changed.connect (on_slider_value_changed);
+        slider.draw_value = false;
+        slider.set_size_request (150, -1);
+
+        add (slider);
+
+        var zoom_in = new Gtk.Image.from_icon_name (Resources.ICON_ZOOM_IN, Gtk.IconSize.MENU);
+        zoom_in_box = new Gtk.EventBox ();
+        zoom_in_box.above_child = true;
+        zoom_in_box.visible_window = false;
+        zoom_in_box.add (zoom_in);
+        zoom_in_box.button_press_event.connect (on_zoom_button_pressed);
+        zoom_in_box.button_release_event.connect (on_zoom_button_released);
+        zoom_in_box.leave_notify_event.connect (() => {
+            clear_zoom_timeouts ();
+
+            return false;
+        });
+
+        add (zoom_in_box);
+    }
```

Comment:
Nearly all of this can be put in the construct clause by creating a slider with null adjustment parameter and then just applying the min, max, step and initial_val in new ():

```

    public ZoomSliderAssembly (double min, double max, double step, double initial_val) {
        slider.set_range (min, max);
        slider.set_increments (step, step);
        zoom_value = initial_val;
    }
```